lib/improver/cli/__init__.py

```diff
@@ -0,0 +1,416 @@
+    # The environment setting has to be done before `import improver.cli`!
+    from ast import literal_eval
+    import os
+    import sys
+
+    if hasattr(obj, 'cli'):
+        return obj
+
+    if not callable(obj):
+        kwargs.pop('helper_class', None)
+        return Clize.get_cli(obj, **kwargs)
+
+    use_clize = os.environ.get('IMPROVER_USE_CLIZE', False)
+    if use_clize:
+        use_clize = literal_eval(use_clize.capitalize())
+
+    legacy_main = getattr(sys.modules[obj.__module__], 'main', None)
+
+    if legacy_main is None or use_clize and obj.__annotations__:
+        # legacy_main is None here for any function defined in this module
+        # before (and including) the main function
+        return Clize.keep(obj, **kwargs)
+
+    def _wrapper(prog: parameters.pass_name, *args):
+        sys.argv[0] = prog.split()[-1]
+        legacy_main(args)
+
+    description = obj.__doc__.split('\n')[0].strip()
+    _obj = Clize.as_is(_wrapper, description=description)
+    obj.cli = _obj.cli
+
+    return obj
+
+
+def clizefy(func=None, helper_class=DocutilizeClizeHelp, **kwargs):
+    """Decorator for creating CLI objects."""
+    from functools import partial
+    if func is None:
+        return partial(clizefy, helper_class=helper_class, **kwargs)
+    func = _clizefy(func, helper_class=helper_class, **kwargs)
+    return func
+
+
+# help command
+
+
+@clizefy(help_names=())
+def improver_help(prog_name: parameters.pass_name,
+                  command=None, *, usage=False):
+    """Show command help."""
+    prog_name = prog_name.split()[0]
+    args = filter(None, [command, '--help', usage and '--usage'])
+    result = execute_command(SUBCOMMANDS_DISPATCHER, prog_name, *args)
+    if not command and usage:
+        result = '\n'.join(line for line in result.splitlines()
+                           if not line.endswith('--help [--usage]'))
+    return result
+
+
+# version command
+
+
+@clizefy(help_names=())
+def improver_version():
+    """Print version"""
+    from improver import __version__
+    return __version__
+
+
+# mapping of command names to CLI objects
+
+
+def _cli_items():
+    """Dynamically discover CLIs."""
+    import importlib
+    import pkgutil
+    from improver.cli import __path__ as improver_cli_pkg_path
+    yield ('help', improver_help)
+    for minfo in pkgutil.iter_modules(improver_cli_pkg_path):
+        mod_name = minfo.name
+        if mod_name != '__main__':
+            mcli = importlib.import_module('improver.cli.' + mod_name)
+            yield (mod_name, clizefy(mcli.process))
+
+
+SUBCOMMANDS_TABLE = OrderedDict(sorted(_cli_items()))
+
+
+# main CLI object with subcommands
+
+
+SUBCOMMANDS_DISPATCHER = clizefy(
+    SUBCOMMANDS_TABLE,
+    description="""IMPROVER NWP post-processing toolbox""",
+    footnotes="""See also improver --help for more information.""")
+
+
+# IMPROVER top level main
+
+
+def brackets_to_nested_list(args):
```

Comment:
Personally, I don't find `brackets_to_nested_list` to be a better name than `unbracket`.

Reply:
I couldn't think of a good one either, but `unbracket` didn't feel descriptive enough - I anticipated some back-and-forth on the name

Reply:
There is a docstring with description and an example that is doctest-able and shows what it does (I know, a true rarity for me 😉). I read the current name as convert brackets to nested list, which isn't an operation that makes sense to me.

Reply:
Fair enough - I appreciated the docstring and doctest/example. I don't mind either way on the name, whatever is the least confusing is fine with me. It's a tiny utility function anyway, that's not (currently intended to be) used externally, so I don't mind if we go back to `unbracket`.